refactor(http): tree-shakable error on JSONP request (#50096)
angular/angular: aio/content/errors/NG2800.md

> @name JSONP support in HttpClient configuration

Reviewer: I think the file should be named as `NG02800.md` (an extra `0` after `NG`), all runtime codes have leading `0` (i.e. `NG0` as a prefix).

> @name Missing HttpClientJsonpModule

Reviewer: ```suggestion
@name Missing JSONP support in HttpClient configuration
```

@category runtime

> @shortDescription HttpClientJsonpModule is required for JSONP support

Reviewer: ```suggestion
@shortDescription Missing JSONP support in HttpClient configuration
```

@description

> You are attempting to make a `JSONP` request without providing the `HttpClientJsonpModule`. 

Reviewer: ```suggestion
Angular produces this error when you try to make a `JSONP` request without providing the necessary support for it in the `HttpClient` configuration.

To enable `JSONP` support, you can:
- either add the `withJsonpSupport()` as an argument during the `provideHttpClient` function call (e.g. `provideHttpClient(withJsonpSupport())`) when `bootstrapApplication` is used
- or import the `HttpClientJsonpModule` in your root AppModule, when NgModule-based bootstrap is used.
```

@debugging

> You are making a JSONP request with the `HttpClient`:
> 
> <code-example format="typescript" language="typescript">
> this.httpClient.jsonp(myUrl, 'callback')
> </code-example>
> 
> To enable `JSONP` support for Angular, provide the `HttpClientJsonpModule` in your `NgModule` or your standalone component. 
> 
> <code-example format="typescript" language="typescript">
> import {provideHttpClient, withJsonpSupport} from  '@angular/common/http';
> 
> &commat;Component({
>   providers: [
>     provideHttpClient(withJsonpSupport()),
>   ]
> })
> </code-example>

Reviewer: ```suggestion
Make sure that the JSONP support is added into your application either by calling the `withJsonpSupport()` function (when the `provideHttpClient()` is used) or importing the `HttpClientJsonpModule` module as described above.
```
